source/transactions/tests/run-command.json

```diff
@@ -0,0 +1,236 @@
+{
+  "database_name": "transaction-tests",
+  "collection_name": "test",
+  "data": [],
+  "tests": [
+    {
+      "description": "run command with default read preference",
+      "operations": [
+        {
+          "name": "startTransaction",
+          "arguments": {
+            "session": "session0"
+          }
+        },
+        {
+          "name": "runCommand",
+          "arguments": {
+            "command": {
+              "count": "test"
+            },
+            "session": "session0"
+          },
+          "result": {
+            "n": 0
+          }
+        },
+        {
+          "name": "commitTransaction",
+          "arguments": {
+            "session": "session0"
+          }
+        }
+      ],
+      "expectations": [
```

Comment:
How is this testing that the correct read preference was used?

Reply:
I thought it was testing it indirectly in that the count and the commitTransaction commands succeed, and that could only happen if run on a primary (since secondary read-only transactions are not supported).  But I just tried it and confirmed that a secondary will happily execute a read-only transaction, so this is not a good test.  @ShaneHarvey does that surprise you?
I suppose the test could be improved by making the command one that has to run on a primary in order to succeed, e.g. insert

Reply:
> But I just tried it and confirmed that a secondary will happily execute a read-only transaction, so this is not a good test. does that surprise you?

Very surprising. Txns on secondaries are not supposed to work AFAIK https://jira.mongodb.org/browse/SERVER-34557.

Can you open a server ticket about it?